encoding.bs

```diff
@@ -94,35 +90,133 @@ bitwise AND by "&amp;", and bitwise OR by "|".
 
 <hr>
 
-<p>A <dfn id=concept-token>token</dfn> is a piece of data, such as a <a>byte</a> or
-<a>scalar value</a>.
+<p>An <dfn id=concept-stream export>I/O queue</dfn> is a type of <a for=/>list</a>
+with <a for=list>items</a> of a particular type (i.e. <a>bytes</a> or <a>scalar
+values</a>). <dfn id="end-of-stream" export>End-of-queue</dfn> is a special
+<a for=list>item</a> that can be present in <a for=/>I/O queues</a> of any type, and that
+signifies no more <a for=list>items</a> are in the queue.
+
+<p>To <dfn for="I/O queue" export>create</dfn> an <a for=/>I/O queue</a>, optionally initialized
+with a list <var>contents</var>, return a new <a for=/>I/O queue</a> containing the
+<a for=list>items</a> of <var>contents</var> in order, if it is given, followed by an
+<a>end-of-queue</a> item.
```

Comment:
Do we really need the create primitive instead of using the shorthand syntax for lists? It seems confusing to me to have it and only allow immediate-queue creation.

Reply:
I was under the impression that the shorthand syntax couldn't be used for list subtypes, but I just checked and saw an example being used for sets, so I guess it's fine.